projects/igniteui-angular/src/lib/slider/slider.component.ts

```diff
@@ -700,6 +719,54 @@ export class IgxSliderComponent implements ControlValueAccessor, OnInit, AfterVi
         );
     }
 
+    private resetValue() {
+        if (!this.isRange) {
+            if (this.value >= this._lowerBound && this.value <= this._upperBound) {
+                this.value = this.value;
+            } else if (this.value < this._lowerBound) {
+                this.value = this._lowerBound;
+            } else if (this.value > this._upperBound) {
+                this.value = this._upperBound;
+            }
+        } else {
+            if ((this.value as IRangeSliderValue).lower >= this._lowerBound &&
```

Comment:
`(this.value as IRangeSliderValue)` is use a lot in this method. It introduces additional complexity to the code. Consider introducing a local variable for this.